lib/memjs/memjs.js

```diff
@@ -84,21 +101,20 @@ Client.create = function(serversStr, options) {
 
 // Chooses the server to talk to by hashing the given key.
 Client.prototype.server = function(key) {
-  // TODO(alevy): should use consistent hashing and/or allow swapping hashing
-  // mechanisms
-  var total = this.servers.length;
-  var origIdx = (total > 1) ? (hashCode(key) % total) : 0;
-  var idx = origIdx;
-  var serv = this.servers[idx];
-  while (serv.wakeupAt &&
-      serv.wakeupAt > Date.now()) {
-    idx = (idx + 1) % total;
-    if (idx === origIdx) {
-      return null;
-    }
-    serv = this.servers[idx];
+  return this.getServerByIndex(this.serverIndex(key));
+};
+
+Client.prototype.serverIndex = function(key) {
+  var total = Object.keys(this.serverMap).length;
+  if (total === 1) {
+    return Object.keys(this.serverMap)[0];
   }
-  return serv;
+
+  return this.hashring.get(key);
+};
+
+Client.prototype.getServerByIndex = function(index) {
+  return this.serverMap[index];
```

Comment:
This is very confusing. It looks like you're taking `index` which my brain infers as `number` and using it as a server hostname.

Reply:
renamed this suite of functions to just these two


Client.prototype.serverKeyToServer = function(serverKey) {
  return this.serverMap[serverKey];
};

Client.prototype.lookupKeyToServerKey = function(key) {
  return this.options.keyToServerHashFunction(this.serverKeys, key);
};